.github/workflows/javadoc-publication.yaml

```diff
@@ -0,0 +1,104 @@
+# yamllint disable rule:line-length
+---
+name: Javadoc publication
+"on":
+  workflow_call:
+    inputs:
+      dryRun:
+        type: boolean
+        description: Flag to enable the dry-run execution
+        default: false
+        required: false
+      javadocDestinationDirectory:
+        type: string
+        description: Name of the directory to use at https://javadocs.sonarsource.org/
+        required: true
+      vaultAddr:
+        type: string
+        description: Custom vault installation
+        default: https://vault.sonar.build:8200
+        required: false
+      artifactoryRoleSuffix:
+        type: string
+        description: artifactory reader suffix specified in vault repo config
+        default: private-reader
+        required: false
+
+jobs:
+  javadoc-publication:
+    name: Publish javadoc
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write  # to authenticate via OIDC
+      contents: read  # to revert a github release
+    timeout-minutes: 5
+    if: ${{ github.event_name == 'release' && github.event.action == 'published' }}
+    outputs:
+      javadoc-publication: ${{ steps.javadoc-publication-output.outcome }}
+    steps:
+      - name: Create local repository directory
+        id: local_repo
+        run: echo "dir=$(mktemp -d repo.XXXXXXXX)" >> $GITHUB_OUTPUT
+      - name: Vault
+        id: secrets
+        uses: SonarSource/vault-action-wrapper@f148a1075c1eff1db5ab474f96c206c5587172d3  # tag=2.5.0-3
+        with:
+          url: ${{ inputs.vaultAddr }}
+          secrets:
+            development/artifactory/token/{REPO_OWNER_NAME_DASH}-${{ inputs.artifactoryRoleSuffix }} access_token  | artifactory_access_token;
+            development/kv/data/slack webhook | slack_webhook;
+            development/aws/sts/javadocs access_key | javadoc_aws_access_key_id;
+            development/aws/sts/javadocs secret_key | javadoc_aws_secret_access_key;
+            development/aws/sts/javadocs security_token | javadoc_aws_security_token;
+      - name: Setup JFrog
+        uses: SonarSource/jfrog-setup-wrapper@92f2e341560f69ccf1763e8efb73f69405545491  # tag=3.2.0-2
```

Comment:
```suggestion
        uses: SonarSource/jfrog-setup-wrapper@117947ea9b62b315c2ee4502ec8d1a6ebade0a4e # 3.2.1
```

Reply:
I agree, however this should be done as part of another PR this is not related to the new feature.